src/sentry/buffer/redis.py

```diff
@@ -288,6 +300,22 @@ def get_hash(
         key = self._make_key(model, field)
         return self._execute_redis_operation(key, RedisOperation.HASH_GET_ALL)
 
+    def process_batch(self, partition: int | None = None) -> None:
+        pending_key = self._make_pending_key(partition)
+        pipe = self.get_redis_connection(pending_key)
+        if not pipe:
+            return
+
+        client = get_cluster_routing_client(self.cluster, self.is_redis_cluster)
+        lock_key = self._lock_key(client, pending_key, ex=60)
```

Comment:
I looked around the Redis docs and can't find what `ex` means, but in `process_pending` we set it to 60 and in `_process_single_incr` it's 10 so I'm guessing ... seconds? And since this task is flushing stuff from the last minute, 60 seconds makes sense, but I could be wrong!

Reply:
`ex` is when this key will expire in redis in seconds. so, this would lock redis for 60s.

I think instead we may want to have it be like `10` as well,  since we shouldn't be spending a lot of time interacting with Redis. We want to make sure it'd  be unlocked before the next celerybeat.